boa3/internal/compiler/filegenerator/filegenerator.py

```diff
@@ -104,51 +109,74 @@ def _methods(self) -> Dict[str, Method]:
 
         :return: a dictionary that maps each method with its identifier
         """
-        from boa3.internal.model.builtin.method import IBuiltinMethod
-        return {name: method for name, method in self._symbols.items()
-                if ((isinstance(method, Method) and method.defined_by_entry)
-                    or (isinstance(method, IBuiltinMethod) and method.is_public)
-                    )}
+        if self._inner_methods is None:
+            from boa3.internal.model.builtin.method import IBuiltinMethod
+            all_entry_file_methods = {}
+            for name, symbol in self._symbols.items():
+                if isinstance(symbol, Method) and symbol.defined_by_entry:
+                    all_entry_file_methods[name] = symbol
+                elif isinstance(symbol, IBuiltinMethod) and symbol.is_public:
+                    all_entry_file_methods[name] = symbol
```

Comment:
Yes, it was done for making the code for readable
Since another elif branch was included, I wanted to make the behavior as clear as possible in the code. If we need to change the identifiers at some point this will be easier to pin point where the changes need to be